### PR TITLE
Support Db2 cdc connector (cp from #450)

### PR DESCRIPTION
**Pull request by ruanhang1993**
Support Db2 cdc connector , cherry pick from #450 .

### CODE REVIEW

File: flink-connector-db2-cdc/src/main/java/com/ververica/cdc/connectors/db2/Db2Source.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+        public DebeziumSourceFunction<T> build() {
+            Properties props = new Properties();
+            props.setProperty("connector.class", Db2Connector.class.getCanonicalName());
+            props.setProperty("database.hostname", checkNotNull(hostname));
+            props.setProperty("database.port", String.valueOf(port));
+            props.setProperty("database.user", checkNotNull(username));
+            props.setProperty("database.password", checkNotNull(password));
+            props.setProperty("database.dbname", checkNotNull(database));
+            props.setProperty("database.server.name", "db2_cdc_source"); // Hard-coded here
```
Comment (GOODBOY008) [inline flink-connector-db2-cdc/src/main/java/com/ververica/cdc/connectors/db2/Db2Source.java:61]: Define a static constant will be better.

File: flink-connector-db2-cdc/src/main/java/com/ververica/cdc/connectors/db2/Db2Source.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            props.setProperty("database.history.skip.unparseable.ddl", String.valueOf(true));
+
+            if (tableList != null) {
+                props.setProperty("table.whitelist", String.join(",", tableList));
+            }
+            if (dbzProperties != null) {
+                dbzProperties.forEach(props::put);
```
Comment (GOODBOY008) [inline flink-connector-db2-cdc/src/main/java/com/ververica/cdc/connectors/db2/Db2Source.java:68]:
```suggestion
                props.putAll(dbzProperties);
```

File: flink-connector-db2-cdc/src/test/java/com/ververica/cdc/connectors/db2/table/Db2ConnectorITCase.java [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+    @ClassRule public static LegacyRowResource usesLegacyRows = LegacyRowResource.INSTANCE;
+
+    @Before
+    public void before() {
+        TestValuesTableFactory.clearAllData();
+        env.setParallelism(1);
+    }
+
+    @Test
+    public void testConsumingAllEvents()
```
Comment (GOODBOY008) [inline flink-connector-db2-cdc/src/test/java/com/ververica/cdc/connectors/db2/table/Db2ConnectorITCase.java:66]: Should add timezone unit test.

File: flink-connector-db2-cdc/src/main/java/com/ververica/cdc/connectors/db2/table/StartupOptions.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+import java.util.Objects;
+
+/** Debezium startup options. */
+public final class StartupOptions {
+    public final StartupMode startupMode;
+
+    /**
+     * Performs an initial snapshot on the monitored database tables upon first startup, and
+     * continue to read the latest logminer.
```
Comment (GOODBOY008) [inline flink-connector-db2-cdc/src/main/java/com/ververica/cdc/connectors/db2/table/StartupOptions.java:27]: This comment should be adjust to db2.

File: flink-connector-db2-cdc/src/test/java/com/ververica/cdc/connectors/db2/Db2TestBase.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+    protected Connection getJdbcConnection() throws SQLException {
+        return DriverManager.getConnection(
+                DB2_CONTAINER.getJdbcUrl(),
+                DB2_CONTAINER.getUsername(),
+                DB2_CONTAINER.getPassword());
+    }
+
+    private static final Path getFilePath(String resourceFilePath) {
```
Comment (GOODBOY008) [inline flink-connector-db2-cdc/src/test/java/com/ververica/cdc/connectors/db2/Db2TestBase.java:95]: Remove `final`.

File: docs/content/connectors/db2-cdc.md [hunk 6]
```diff
@@ -0,0 +1,369 @@
+# Db2 CDC Connector
```
Comment (GOODBOY008) [inline docs/content/connectors/db2-cdc.md:1]: `flink-cdc-connectors\docs\content\about.md` shoud be updated too and miss `quickstart`.

**Review by GOODBOY008 — COMMENTED** [review]
@ruanhang1993 Thanks for your contributions. Should add `e2e` test and `metadata columns` support.

File: flink-connector-db2-cdc/src/test/java/com/ververica/cdc/connectors/db2/table/Db2ConnectorITCase.java (hunk 3 above)
Comment (ruanhang1993) [reply flink-connector-db2-cdc/src/test/java/com/ververica/cdc/connectors/db2/table/Db2ConnectorITCase.java:66]: Actually the time zone information will not be contained in the debezium db2 connector. IMO this time zone setting may be useless.

File: flink-connector-db2-cdc/src/test/java/com/ververica/cdc/connectors/db2/Db2TestBase.java [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+    protected static final Db2Container DB2_CONTAINER =
+            new Db2Container(DEBEZIUM_DOCKER_IMAGE_NAME)
+                    .withDatabaseName("testdb")
+                    .withUsername("db2inst1")
+                    .withPassword("flinkpw")
+                    .withEnv("AUTOCONFIG", "false")
+                    .withEnv("ARCHIVE_LOGS", "true")
+                    .acceptLicense()
+                    .withLogConsumer(logConsumer)
```
Comment (GOODBOY008) [inline flink-connector-db2-cdc/src/test/java/com/ververica/cdc/connectors/db2/Db2TestBase.java:61]: Simply to `withLogConsumer(new Slf4jLogConsumer(LOG))`

**Review by GOODBOY008 — COMMENTED** [review]
@ruanhang1993  LGTM , @leonardBang Pls cc.

**Comment by GOODBOY008** [comment]
@ruanhang1993 DB2 unit test is not runned , you can modify file in directory `tools/azure-pipelines` and `tools/ci`.

**Review by leonardBang — APPROVED** [review]
Thanks @ruanhang1993 and @GOODBOY008 for the great work. LGTM, I've checked the tutorial it works well.

**Comment by leonardBang** [comment]
implemented in 3592dcc4ee7bc6891e68dbdf2a02b82d6b098f40

**Pull request closed without merging**